### PR TITLE
Ds/implement lifting operator

### PR DESCRIPTION
**Pull request by dsteinmo**

### CODE REVIEW

**Comment by dsteinmo** [comment]
@k7miller Any input on this one? I'm hoping to merge soon and move onto a solver or seeing what's left in the way of a solver

File: src/TriangleNodesProvisioner.cpp [hunk 1]
```diff
@@ -288,4 +291,157 @@ namespace blitzdg {
...
+        std::cout << faceR << std::endl;
+
+
+        Vandermonde.computeVandermondeMatrix(faceR, V1D, V1Dinv);
+        massEdgeInv = sum(V1D(ii,kk)*V1D(jj,kk), kk);
+
+        Inverter.computeInverse(massEdgeInv, massEdge2);
+        for (index_type i=0; i < NumFacePoints; ++i) {
+            for (index_type j=NumFacePoints; j < 2*NumFacePoints; ++j) {
+                E(Fm(i,1),j) = massEdge2(i,j % NumFacePoints);
```
Comment (k7miller) [inline src/TriangleNodesProvisioner.cpp]: If j ranges from `NumFacePoints` to `2*NumFacePoints - 1`, then `j % NumFacePoints` ranges from `0` to `NumFacePoints - 1`, so you can replace `j % NumFacePoints` with `j - NumFacePoints`.

File: src/TriangleNodesProvisioner.cpp [hunk 2]
```diff
@@ -288,4 +291,157 @@ namespace blitzdg {
...
+
+        std::cout << faceS << std::endl;
+
+        Vandermonde.computeVandermondeMatrix(faceS, V1D, V1Dinv);
+        massEdgeInv = sum(V1D(ii,kk)*V1D(jj,kk), kk);
+
+        Inverter.computeInverse(massEdgeInv, massEdge3);
+        for (index_type i=0; i < NumFacePoints; ++i) {
+            for (index_type j=2*NumFacePoints; j < 3*NumFacePoints; ++j) {
+                E(Fm(i,2),j) = massEdge3(i,j % NumFacePoints);
```
Comment (k7miller) [inline src/TriangleNodesProvisioner.cpp]: Replace `j % NumFacePoints` with `j - 2 * NumFacePoints`, similar to the comment above.

File: src/TriangleNodesProvisioner.cpp [hunk 3]
```diff
@@ -288,4 +291,157 @@ namespace blitzdg {
...
+        std::cout << "Minv:" << std::endl << MassInv << std::endl;
+
+        // Multiply by inverse mass matrix;
+        Liftref = sum(MassInv(ii,kk)*E(kk,jj),kk);
+
+        std::cout << "Lift:" << std::endl << Liftref << std::endl;
+    }
+
+    const real_matrix_type & TriangleNodesProvisioner::get_Lift() const {
+        return *Lift.get();
```
Comment (k7miller) [inline src/TriangleNodesProvisioner.cpp]: A unique_ptr has very similar semantics to a pointer, so you should be able to return `*Lift`, which is a bit cleaner.

**Review by k7miller — APPROVED** [review]
Code looks good to me buddy, just a few minor comments. Nice work!

**Pull request merged**